src/themes/utils/generateTheme.ts

```diff
@@ -63,17 +75,29 @@ export default async function generateTheme(themeName: string): Promise<void> {
   const structure = await readFileSync(getFile(themeName, THEME_GENERATOR_FILE)).toString()
   const values = await readFileSync(getFile(themeName, PRIMITIVES_FILE)).toString()
 
+  const resolvedTheme = resolveThemeTokens(structure, values)
+
+  writeFileSync(getFile(themeName, GENERATED_FILE), JSON.stringify(resolvedTheme, null, 2))
+}
+
+function resolveThemeTokens(structure: string, values: string): Theme {
   const themeStructure: object = JSON.parse(structure)
   const themeValues: Theme = JSON.parse(values)
 
-  const resolveVariable = resolveThemeValues(themeValues)
+  const resolveVariable = resolveThemeValues(themeValues, themeStructure)
 
-  const resolvedTheme: Theme = traverse(themeStructure).map(function(node) {
+  return traverse(themeStructure).map(function(node) {
     const isLeaf = node?.$value && node?.$type
-    const resolvedNode = resolveVariable(isLeaf ? node.$value : node)
 
-    this.update(resolvedNode)
+    if (!isLeaf) {
+      this.update(node)
```

Comment:
why isn't the `resolveVariable` invoked over `node` as in the previous implementation?

Reply:
because there is no value to resolve, but previously the function returned the same node